diff --git a/examples/cpu/inference/python/llm/Dockerfile b/examples/cpu/inference/python/llm/Dockerfile
index 41e29c42f..1bb984965 100644
--- a/examples/cpu/inference/python/llm/Dockerfile
+++ b/examples/cpu/inference/python/llm/Dockerfile
@@ -36,7 +36,7 @@ RUN useradd -m ubuntu && \
 USER ubuntu
 WORKDIR /home/ubuntu
 
-RUN curl -fsSL -v -o miniconda.sh -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+RUN curl -fsSL -v -o miniconda.sh -O https://github.com/conda-forge/miniforge/releases/download/24.1.2-0/Miniforge3-24.1.2-0-Linux-x86_64.sh  && \
     bash miniconda.sh -b -p ./miniconda3 && \
     rm miniconda.sh && \
     echo "source ~/miniconda3/bin/activate" >> ./.bashrc
@@ -64,4 +64,5 @@ RUN . ./miniconda3/bin/activate && \
     conda clean -a -y && \
     sudo mv ./oneCCL_release /opt/oneCCL && \
     sudo chown -R root:root /opt/oneCCL && \
-    sed -i "s|ONECCL_PATH=.*|ONECCL_PATH=/opt/oneCCL|" ./tools/env_activate.sh
+    sed -i "s|ONECCL_PATH=.*|ONECCL_PATH=/opt/oneCCL|" ./tools/env_activate.sh && \
+    pip install "numpy<2.0.0" "setuptools<81"
diff --git a/examples/cpu/inference/python/llm/distributed/run_generation_with_deepspeed.py b/examples/cpu/inference/python/llm/distributed/run_generation_with_deepspeed.py
index f58527c1a..f12f9b048 100644
--- a/examples/cpu/inference/python/llm/distributed/run_generation_with_deepspeed.py
+++ b/examples/cpu/inference/python/llm/distributed/run_generation_with_deepspeed.py
@@ -186,10 +186,10 @@ world_size = get_int_from_env(["WORLD_SIZE", "PMI_SIZE"], "1")
 deepspeed.init_distributed(get_accelerator().communication_backend_name())
 
 
-def print_rank0(*msg):
+def print_rank0(*msg, **kwargs):
     if local_rank != 0:
         return
-    print(*msg)
+    print(*msg, **kwargs)
 
 
 # Model loading and instantiating on GPUs
@@ -508,6 +508,7 @@ def generate():
 
 
 def trace_handler(prof):
+    prof.export_chrome_trace(f"/home/ubuntu/.cache/trace_{local_rank}.json")
     print(prof.key_averages().table(sort_by="self_cpu_time_total", row_limit=-1))
 
 
@@ -540,6 +541,10 @@ else:
             activities=[torch.profiler.ProfilerActivity.CPU],
             schedule=torch.profiler.schedule(wait=1, warmup=3, active=1),
             on_trace_ready=trace_handler,
+            with_modules=True,
+            with_stack=True,
+            record_shapes=True,
+            profile_memory=True,
         ) as prof:
             for i in range(5):
                 gen_ids, outputs = generate()
@@ -574,3 +579,10 @@ else:
         print_rank0("Average 2... latency: %.3f sec." % average_2n_latency)
         print_rank0("P90 2... latency: %.3f sec." % p90_latency)
         print_rank0("P99 2... latency: %.3f sec." % p99_latency)
+        print_rank0("token times:")
+        for ta in total_list:
+            print_rank0(ta[0], end="")
+            for t in ta[1:]:
+                print_rank0(", ", end=str(t))
+            print_rank0()
+        print_rank0("end token times")
\ No newline at end of file
diff --git a/examples/cpu/inference/python/llm/single_instance/run_generation.py b/examples/cpu/inference/python/llm/single_instance/run_generation.py
index 999f81c04..d6598c1be 100644
--- a/examples/cpu/inference/python/llm/single_instance/run_generation.py
+++ b/examples/cpu/inference/python/llm/single_instance/run_generation.py
@@ -240,3 +240,10 @@ if args.benchmark:
         print("Average 2... latency: %.3f sec." % average_2n_latency)
         print("P90 2... latency: %.3f sec." % p90_latency)
         print("P99 2... latency: %.3f sec." % p99_latency)
+        print("token times:")
+        for ta in total_list:
+            print(ta[0], end="")
+            for t in ta[1:]:
+                print(", ", end=str(t))
+            print()
+        print("end token times")
\ No newline at end of file
diff --git a/examples/cpu/inference/python/llm/single_instance/run_quantization.py b/examples/cpu/inference/python/llm/single_instance/run_quantization.py
index a351bc7bf..599974a81 100644
--- a/examples/cpu/inference/python/llm/single_instance/run_quantization.py
+++ b/examples/cpu/inference/python/llm/single_instance/run_quantization.py
@@ -755,3 +755,10 @@ if args.benchmark:
         print("Average 2... latency: %.3f sec." % average_2n_latency)
         print("P90 2... latency: %.3f sec." % p90_latency)
         print("P99 2... latency: %.3f sec." % p99_latency)
+        print("token times:")
+        for ta in total_list:
+            print(ta[0], end="")
+            for t in ta[1:]:
+                print(", ", end=str(t))
+            print()
+        print("end token times")
diff --git a/examples/cpu/inference/python/llm/tools/env_setup.sh b/examples/cpu/inference/python/llm/tools/env_setup.sh
index 92112a009..9787a1cb4 100644
--- a/examples/cpu/inference/python/llm/tools/env_setup.sh
+++ b/examples/cpu/inference/python/llm/tools/env_setup.sh
@@ -83,7 +83,8 @@ if [ $((${MODE} & 0x02)) -ne 0 ]; then
     fi
 
     # Install deps
-    conda install -y cmake ninja
+    conda install -y -c conda-forge "cmake>=3.5,<4"
+    conda install -y ninja
 
     echo "#!/bin/bash" > ${AUX_INSTALL_SCRIPT}
     if [ $((${MODE} & 0x04)) -ne 0 ]; then
@@ -97,9 +98,11 @@ if [ $((${MODE} & 0x02)) -ne 0 ]; then
             exit 4
         else
             echo "python -m pip install torch==${VER_TORCH} --index-url https://download.pytorch.org/whl/cpu" >> ${AUX_INSTALL_SCRIPT}
-            echo "python -m pip install intel-extension-for-pytorch==${VER_IPEX} oneccl-bind-pt==${VER_TORCHCCL} --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/cpu/us/" >> ${AUX_INSTALL_SCRIPT}
+            echo "python -m pip install intel-extension-for-pytorch==${VER_IPEX%+cpu}" >> ${AUX_INSTALL_SCRIPT}
+            echo "python -m pip install oneccl-bind-pt==${VER_TORCHCCL} --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/cpu/us/" >> ${AUX_INSTALL_SCRIPT}
             python -m pip install torch==${VER_TORCH} --index-url https://download.pytorch.org/whl/cpu
-            python -m pip install intel-extension-for-pytorch==${VER_IPEX} oneccl-bind-pt==${VER_TORCHCCL} --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/cpu/us/
+            python -m pip install intel-extension-for-pytorch==${VER_IPEX%+cpu}
+            python -m pip install oneccl-bind-pt==${VER_TORCHCCL} --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/cpu/us/
         fi
     else
         function ver_compare() {
@@ -191,8 +194,8 @@ if [ $((${MODE} & 0x02)) -ne 0 ]; then
 
     # Install DeepSpeed
     if [ $((${MODE} & 0x04)) -ne 0 ]; then
-        echo "python -m pip install deepspeed==${VER_DS_SYCL}" >> ${AUX_INSTALL_SCRIPT}
-        python -m pip install deepspeed==${VER_DS_SYCL}
+        echo "python -m pip install --no-build-isolation deepspeed==${VER_DS_SYCL}" >> ${AUX_INSTALL_SCRIPT}
+        python -m pip install --no-build-isolation deepspeed==${VER_DS_SYCL}
     else
         if [ -d DeepSpeed ]; then
             rm -rf DeepSpeed
